opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -29,12 +43,41 @@ impl<T: Number<T>> Default for ValueMap<T> {
 
 impl<T: Number<T>> ValueMap<T> {
     fn new() -> Self {
+        let buckets = std::iter::repeat_with(|| Mutex::new(None))
+            .take(BUCKET_COUNT)
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap();
```

Comment:
if we are never expecting to reach unwrap(), add a code comment. I am not sure if there is existing precedence in this repo for that.

Reply:
Oh, [panic](https://github.com/open-telemetry/opentelemetry-rust/pull/1564#discussion_r1506756305) has been removed, unwrap() will be always called to convert the vector into an array reference.

Reply:
Modified the code to handle error from unwrap().